ste/mgr-JobPartMgr.go

```diff
@@ -111,6 +111,7 @@ func NewClientOptions(retry policy.RetryOptions, telemetry policy.TelemetryOptio
 		perRetryPolicies = append(perRetryPolicies, NewSourceAuthPolicy(srcCred))
 	}
 	retry.ShouldRetry = GetShouldRetry(&log)
+	retry.StatusCodes = GetRetryStatusCodes(&log)
```

Comment:
Why are we adding this variable? We don't want to retry on every retry status code key the user provides, we only want to retry if the status code and error code both match (or just status code if there is no error code shared)